portal-web/test/functional/com/liferay/portalweb/macros/headless/journal/HeadlessWebcontentFolderAPI.macro

```diff
@@ -0,0 +1,195 @@
+
+		if (!(isSet(userEmailAddress))) {
+			var userEmailAddress = "test@liferay.com";
+		}
+
+		if (!(isSet(userPassword))) {
+			var userPassword = "test";
+		}
+
+		if (!(isSet(description))) {
+			var description = "";
+		}
+
+		if (!(isSet(externalReferenceCode))) {
+			var externalReferenceCode = "";
+		}
+
+		if (!(isSet(name))) {
+			var name = "";
+		}
+
+		if (!(isSet(viewableBy))) {
+			var viewableBy = "Anyone";
+		}
+
+		var curl = '''
+			${portalURL}/o/headless-delivery/v1.0/structured-content-folders/${parentStructuredContentFolderId}/structured-content-folders \
+				-u ${userEmailAddress}:${userPassword} \
+				-H accept: application/json \
+				-H Content-Type: application/json \
+				-d {
+					"description": "${description}",
+                    "externalReferenceCode": "${externalReferenceCode}",
+                    "name": "${name}",
+					"viewableBy": "${viewableBy}"
+				}
+		''';
+
+		var response = JSONCurlUtil.post("${curl}");
+
+		return "${response}";
+	}
+
+	macro _addWebcontentFolder {
+		Variables.assertDefined(parameterList = "${name}");
+
+		var portalURL = JSONCompany.getPortalURL();
+
+		if (!(isSet(userEmailAddress))) {
+			var userEmailAddress = "test@liferay.com";
+		}
+
+		if (!(isSet(userPassword))) {
+			var userPassword = "test";
+		}
+
+		if (!(isSet(description))) {
+			var description = "";
+		}
+
+		if (!(isSet(externalReferenceCode))) {
+			var externalReferenceCode = "";
+		}
+
+		if (!(isSet(name))) {
+			var name = "";
+		}
+
+		if (!(isSet(viewableBy))) {
+			var viewableBy = "Anyone";
+		}
+
+		if (isSet(assetLibraryId)) {
+			var api = "asset-libraries/${assetLibraryId}/structured-content-folders";
+		}
+		else {
+			var api = "sites/${siteId}/structured-contents";
+		}
+
+		var curl = '''
+			${portalURL}/o/headless-delivery/v1.0/${api} \
+				-u ${userEmailAddress}:${userPassword} \
+				-H accept: application/json \
+				-H Content-Type: application/json \
+				-d {
+					"description": "${description}",
+                    "externalReferenceCode": "${externalReferenceCode}",
+                    "name": "${name}",
+                    "parentStructuredContentFolderId": 0,
+					"viewableBy": "${viewableBy}"
+				}
+		''';
+
+		var response = JSONCurlUtil.post("${curl}");
+
+		return "${response}";
+	}
+
+	macro _getStructuredContentFoldersByErcInAssetLibrary {
+		Variables.assertDefined(parameterList = "${assetLibraryId},${externalReferenceCode}");
+
+		var portalURL = JSONCompany.getPortalURL();
+
+		var curl = '''
+            ${portalURL}/o/headless-delivery/v1.0/asset-libraries/${assetLibraryId}/structured-content-folders/by-external-reference-code/${externalReferenceCode} \
+            -u test@liferay.com:test \
+            -H Content-Type: application/json
+        ''';
+
+		var curl = JSONCurlUtil.get("${curl}");
+
+		return "${curl}";
+	}
+
+	macro assertExternalReferenceCodeWithCorrectValue {
+		if (!(isSet(expectedExternalReferenceCodeValue))) {
```

Comment:
why do we go to check mysql? Isn't there a GET request to find out the External Reference code value?

Reply:
Oh, this is because I haven't found "uuid" in the response using GET requests in StructuredContentFolder. This way only works on mysql database. Any suggestions?

Reply:
which step exactly are we making this assertion for?

Reply:
we need to speak to the PO then about whether the field is missing intentionally or it should appear. It's kinda strange that it appears for structured content but not for the folder.

Reply:
the problem I see here is that we are using different databases in different environments, and so the request you're submitting may not always be accurate for all the envs.